### ISSUE TITLE
checkSourceRootExistence throw exception

### ISSUE THREAD

**Issue by zyq8709** [oracle/opengrok#2657]
I tried to setup opengrok under ubuntu 18 and tomcat 8. I specified the source directory which was on a  mounted NTFS partition. I tested the access permission of that directory running a program under tomcat8 uid and it succeeded. However, while I opened the web page, it showed the exception information which is similar to the issue https://github.com/oracle/opengrok/issues/1983 . I don't think https://github.com/oracle/opengrok/blob/81b586e651d9f2a40ab5a985d2b7080ae732dd83/opengrok-indexer/src/main/java/org/opengrok/indexer/web/PageConfig.java#L1661 should fail because of my test. Could you help me find the reason? Thanks!

**Comment by tulinkry** [comment]
How does your indexer command look like? Can you show us the source root as you retrieve `curl http://localhost:8080/source/api/v1/configuration` (replace `http://localhost:8080/source/` with your correct url to the opengrok webapp)?

**Comment by tulinkry** [comment]
Did you run indexer at all?

**Comment by zyq8709** [comment]
Sorry and thanks for your kind help! I forgot to check the issue.  I deleted all previous work and have resumed the issue. Here is the result from the api:
<?xml version="1.0" encoding="UTF-8"?>
<java version="10.0.2" class="java.beans.XMLDecoder">
 <object class="org.opengrok.indexer.configuration.Configuration" id="Configuration0">
  <void property="cmds">
   <object class="java.util.Collections" method="unmodifiableMap">
    <object class="java.util.HashMap">
     <void method="put">
      <string>org.opengrok.indexer.history.ClearCaseRepository</string>
      <string>cleartool</string>
     </void>
     <void method="put">
      <string>org.opengrok.indexer.history.PerforceRepository</string>
      <string>p4</string>
     </void>
    </object>
   </object>
  </void>
  <void property="ctags">
   <string>/usr/local/bin/ctags</string>
  </void>
  <void property="dataRoot">
   <string>/media/*****/***********/opengrok/data</string>
  </void>
  <void id="IgnoredNames0" property="ignoredNames">
   <void id="IgnoredDirs0" property="ignoredDirs">
    <void property="items">
     <void method="add">
      <string>.bk</string>
     </void>
     <void method="add">
      <string>.hg</string>
     </void>
     <void method="add">
      <string>.bzr</string>
     </void>
     <void method="add">
      <string>.git</string>
     </void>
     <void method="add">
      <string>.svn</string>
     </void>
     <void method="add">
      <string>SCCS</string>
     </void>
     <void method="add">
      <string>.razor</string>
     </void>
     <void method="add">
      <string>RCS</string>
     </void>
     <void method="add">
      <string>CVS</string>
     </void>
     <void method="add">
      <string>CVSROOT</string>
     </void>
     <void method="add">
      <string>.repo</string>
     </void>
    </void>
   </void>
   <void id="IgnoredFiles0" property="ignoredFiles">
    <void property="items">
     <void method="add">
      <string>.hgtags</string>
     </void>
     <void method="add">
      <string>.hgignore</string>
     </void>
     <void method="add">
      <string>.gitignore</string>
     </void>
     <void method="add">
      <string>.p4config</string>
     </void>
     <void method="add">
      <string>.cvsignore</string>
     </void>
    </void>
   </void>
  </void>
  <void property="sourceRoot">
   <string>/media/*****/***********/WebKit/Source</string>
  </void>
  <void property="tagsEnabled">
   <boolean>true</boolean>
  </void>
 </object>
</java>
BTW: if I index the code with the paths in the internal file system, it works fine.

**Comment by vladak** [comment]
Are you able to access the same path on command line ?

**Comment by vladak** [comment]
The https://docs.oracle.com/javase/7/docs/api/java/io/File.html#exists() will basically do `open()` and/or `stat()` syscalls on the path in question. What happens if you run stat(1) on the path ?

**Comment by zyq8709** [comment]
Yes, I also checked the code which implements exists(). As I said, I wrote a program invoking the above syscalls and ran it under tomcat8 uid, and it succeeded. It seems very weird to me as well.

**Comment by Vigi0303** [comment]
Any progress here? As if I'm facing the same problem.

I install opengrok and tomcat8 on the sda disk which was the root mount point, and specify the src dir to _/home/xxx/xxx_ as the sdb disk was mount as /home.

I get the error:
```
There was an error!
The source root specified in your configuration does not point to a valid directory! Please configure your webapp.
Source root path "/home/xxx/xxx" does not exist
```

As I change the src dir to path on sda disk like /var/lib/opengrok, it works fine.
It doesn't make sense.

**Comment by Vigi0303** [comment]
Ignore the comment above.
After some test, I figured out that's causing by fs permission.
Tomcat8 service runs as user tomcat8 by default, and it doesn't have read permission to path _/home/xxx/xxx_. It doesn't matter with the different disk.
